### PR TITLE
Tim nolet/sc 10422/gh sync docs v1

### PR DESCRIPTION
**Pull request by tnolet**
## Affected Components
* [ ] Content & Marketing
* [ ] Test
* [x] Docs
* [ ] Learn
* [ ] Other

## Pre-Requisites
* [ ] Code is linted (`npm run lint`)

### CODE REVIEW

File: site/content/docs/browser-checks/github-code-sync.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+2. Type out the path to the entry file you want to sync, just like the "Sync on Commit" strategy.
+
+Now you need to make sure you have integrated one of our supported integrations that can tell Checkly a deployment is done. 
+Let's use the [Vercel](/docs/cicd/vercel/) integration as an example.
+
+After setting up the Vercel integration, two workflows open up.
+
+**1. Working on Preview branches**
+
+You push an updated app to a development branch. Vercel proceeds to deploy a `Preview` environment and Checkly triggers
```
Comment (umutuzgur) [inline site/content/docs/browser-checks/github-code-sync.md:82]: This is actually called `staging` on the vercel side

File: site/content/docs/browser-checks/github-code-sync.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+the checks (or check groups) after syncing your code. Your checks pass, or fail. We do not store the synced check code yet,
+as you marked `Production` as your target environment: we only use the synced check code for checks triggered by the deployment to `Preview`.
```
Comment (umutuzgur) [inline site/content/docs/browser-checks/github-code-sync.md:84]: It is lowercase at vercel as `production`

File: site/content/docs/browser-checks/github-code-sync.md (hunk 1 above)
Comment (tnolet) [reply site/content/docs/browser-checks/github-code-sync.md:82]: @umutuzgur hmmm, I don't think a user is exposed to that information. They only see `Preview` and `Production` 
CleanShot 2022-08-04 at 11 19 41@2x

Comment (umutuzgur) [reply site/content/docs/browser-checks/github-code-sync.md:82]: This is a reason why I thought about having a sync per integration. So we can make it look nicer and get rid of some complexity for customers

**Comment by stefanjudis** [comment]
@tnolet That's a solid start thanks!

I'd like to change the structure a little restructure the three concepts. They're actually only two: 
- what to sync
- when to sync?

I'll move forward with this adjusted outline if that works.

```
Syncing code from Github

## Linking files from a GitHub repo
## File sync strategies
### Sync on commit
### Sync on deployment
## Working with local dependencies
## Working with external dependencies
```

**Comment by tnolet** [comment]
> @tnolet That's a solid start thanks!
> 
> I'd like to change the structure a little restructure the three concepts. They're actually only two:
> 
> * what to sync
> * when to sync?
> 
> I'll move forward with this adjusted outline structure (wording not final) if that works.
> 
> ```
> Syncing code from Github
> 
> ## Linking files from a GitHub repo
> ## File sync strategies
> ### Sync on commit
> ### Sync on deployment
> ## Working with local dependencies
> ## Working with external dependencies
> ```

@stefanjudis please go ahead!

File: site/content/docs/browser-checks/github-code-sync.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+In the GitHub integration section...
+
+1. Toggle **Sync from GitHub**.
+2. Pick a repository with the file you want to sync.
+3. You will now choose your entry file and syncing strategy.
+
+> Note that your **entry file** can `require` other local file dependencies. Checkly will fetch them recursively.
+See [Working with local dependencies](#working-with-local-dependencies) below for more details.
+
```
Comment (stefanjudis) [inline site/content/docs/browser-checks/github-code-sync.md:40]: Note to myself, document what happens when file are moved or deleted.

**Review by stefanjudis — APPROVED** [review]
@tnolet Good job!

File: site/content/docs/browser-checks/github-code-sync.md (hunk 2 above)
Comment (stefanjudis) [reply site/content/docs/browser-checks/github-code-sync.md]: I just triple checked this and `production` is not synced when deploying a repo with vercel (without Vercel integration). 

image

image

`Production` works on the other hand. :)

**Pull request merged**